pkg/trie/triedb/lookup.go

```diff
@@ -6,64 +6,64 @@ package triedb
 import (
 	"bytes"
 
-	"github.com/ChainSafe/gossamer/lib/common"
-	"github.com/ChainSafe/gossamer/pkg/trie/cache"
 	"github.com/ChainSafe/gossamer/pkg/trie/db"
 	"github.com/ChainSafe/gossamer/pkg/trie/triedb/codec"
+	"github.com/ChainSafe/gossamer/pkg/trie/triedb/hash"
+	"github.com/ChainSafe/gossamer/pkg/trie/triedb/nibbles"
 )
 
-type TrieLookup struct {
+// Description of what kind of query will be made to the trie.
+type Query[Item any] func(data []byte) Item
+
+type TrieLookup[H hash.Hash, Hasher hash.Hasher[H], QueryItem any] struct {
 	// db to query from
 	db db.DBGetter
 	// hash to start at
-	hash common.Hash
-	// cache to speed up the db lookups
-	cache cache.TrieCache
-	// Optional recorder for recording trie accesses
-	recorder *Recorder
+	hash H
+	// optional cache to speed up the db lookups
+	cache Cache
+	// optional recorder for recording trie accesses
+	recorder TrieRecorder
 }
 
-func NewTrieLookup(db db.DBGetter, hash common.Hash, cache cache.TrieCache, recorder *Recorder) TrieLookup {
-	return TrieLookup{
+func NewTrieLookup[H hash.Hash, Hasher hash.Hasher[H], QueryItem any](
```

Comment:
Where are you using the `QueryItem`? 🤔

Reply:
Whoops! 😅 This is part of the caching changes coming next.  I'll remove on this branch.